contracts/AgentFactory.sol

```diff
@@ -28,12 +36,14 @@ contract AgentFactory is GenericManager {
     }
 
     /// @dev Creates agent.
+    /// @param mechMarketplace Mech marketplace address.
     /// @param agentOwner Owner of the agent.
     /// @param agentHash IPFS CID hash of the agent metadata.
     /// @param price Minimum required payment the agent accepts.
     /// @return agentId The id of a created agent.
     /// @return mech The created mech instance address.
     function create(
+        address mechMarketplace,
```

Comment:
Pass a specific marketplace in mech creation to associate with.

Reply:
Just for generality, imo it should be optional address param, and last param, so that when no param is passed the mech gets created and not registered on market place.

Reply:
Shall we revert then if the request goes not via the marketplace, or the marketplace is still not defined? Or we need to support mechs working without marketplaces?

Reply:
We should support mechs that are separate from a marketplace, as before, imo.